src/algos/simulation.jl

```diff
@@ -38,23 +36,39 @@ function simulation(model::AbstractNumericModel, dr::AbstractDecisionRule,
 
     for t in 1:horizon
         s = copy(view(s_simul, :, :, t))
-        m = copy(view(epsilons,:,:,t))
+        m = copy(view(epsilons, :, :, t))
         x = dr(m,s)
         x_simul[:, :, t] = x
         if t < horizon
-          M = view(epsilons,:,:,t+1)
+          M = view(epsilons, :, :, t+1)
           ss = view(s_simul, :, :, t+1)
-          ss = Dolo.transition!(model, vec(ss), vec(m), vec(s), vec(x), vec(M), params)
+          println([size(e) for e in [ss,m,s,x,M]])
+          ss = Dolo.transition!(model, (ss), (m), (s), (x), (M), params)
           s_simul[:, :, t+1] = ss
         end
     end
 
-    return cat(2, s_simul, x_simul)::Array{Float64,3}
+    return cat(2, epsilons, s_simul, x_simul)::Array{Float64,3}
+
+end
+
+function simulate(model::AbstractNumericModel, dr::AbstractDecisionRule, s0::AbstractVector; kwargs...)
+    e0 = model.calibration[:exogenous]
+    return simulate(model, dr, s0, e0; kwargs...)
+end
 
+function simulate(model::AbstractNumericModel,  dr::AbstractDecisionRule; kwargs...)
+    s0 = model.calibration[:states]
+    e0 = model.calibration[:exogenous]
```

Comment:
So the default starting point for all states (exo or endorsed) should be the value into the calibration supplied with the model which is what model.calibration contains. For exogenous values the default value is 0 (for now, it should default to the default value of the exogenous process)